### ISSUE TITLE
`multiHilbertPolynomial` in `CorrespondenceScrolls` gives weird results

### ISSUE THREAD

**Issue by jkyang92** [Macaulay2/M2#1344]
`multiHilbertPolynomial` in `CorrespondenceScrolls` appears to be incorrect or perhaps I'm misunderstanding what it's supposed to compute. Additionally, it is writing to user variables.

```
needsPackage "CorrespondenceScrolls";

S = productOfProjectiveSpaces{1,2}

I = ideal(x_(0,0),x_(1,0));

M = S^1/I;
g = multiHilbertPolynomial(M);

a -- multiHilbertPolynomial is writing to User variables!

--numerical values seem very wrong
print hilbertFunction({10,10},M); -- 11
print substitute(g,{s_0=>10,s_1=>10}); -- -398024

--compute the hilbert polynomial via resolutions.
print multigraded betti res M;
h = multiHilbertPolynomial(S^1);
g' = h-substitute(h,{s_0=>s_0-1})-substitute(h,{s_1=>s_1-1})+substitute(h,{s_0=>s_0-1,s_1=>s_1-1});

print substitute(g',{s_0=>10,s_1=>10}); -- 11

```

**Comment by DanGrayson** [comment]
@ foschreyer

**Comment by DanGrayson** [comment]
fixed in 2999781d6a1039b7b50b8c7817f1a3e07a3e1429 and ba3229e22b766e03ad36b2b39d6bdcee79648cab
